lib/my_obfuscate.rb

```diff
@@ -92,18 +98,14 @@ def missing_column_list(table_name, columns)
   end
 
   def check_for_table_columns_not_in_definition(table_name, columns)
-    unless unspecified_columns_behavior == :ignore
-      missing_columns = missing_column_list(table_name, columns)
-      unless missing_columns.length == 0
-        error_message = missing_columns.map do |missing_column|
-          "Column '#{missing_column}' defined in table '#{table_name}', but not found in table definition, please fix your obfuscator config."
-        end.join("\n")
-        if unspecified_columns_behavior == :fail
-          raise RuntimeError.new(error_message)
-        else
-          STDERR.puts(error_message)
-        end
-      end
+    missing_columns = missing_column_list(table_name, columns)
+
+    unless missing_columns.empty?
+      error_message = missing_columns.map do |missing_column|
+        "Column '#{missing_column}' defined in table '#{table_name}', but not found in table definition, please fix your obfuscator config."
+      end.join("\n")
```

Comment:
Move join + split logic into the error handler for specifically the case where we are failing fast. There's no need to join when in `:warn` or `:ignore` mode.